File: app/src/main/java/com/example/task2/javafolder/viewviewmodel/fragments/startingfragments/collectionsstarting/CollectionsStartingFragment.java
package com.example.task2.javafolder.viewviewmodel.fragments.startingfragments.collectionsstarting;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProvider;

import com.example.task2.R;
import com.example.task2.javafolder.viewviewmodel.fragments.BaseViewModel;
import com.example.task2.javafolder.viewviewmodel.fragments.parentfragment.collections.CollectionsViewModel;
import com.example.task2.javafolder.viewviewmodel.fragments.startingfragments.ParentStartingFragment;
import com.example.task2.javafolder.viewviewmodel.fragments.startingfragments.ParentStartingViewModel;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class CollectionsStartingFragment extends ParentStartingFragment {
    private CollectionsStartingViewModel viewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewModel = new ViewModelProvider(this).get(CollectionsStartingViewModel.class);
    }

    @Override
    public BaseViewModel determineViewModelProvider() {
        return viewModel;
    }

    @Override
    public int determineNextFragment() {
        return R.id.action_collectionsStartingFragment_to_fragmentWithCollectionsBenchmarks;
    }

    @Override
    public int stringIdForTextView() {
        return R.string.enter_collection_size;
    }

    @Override
    public int stringIdForEditTextHint() {
        return R.string.edit_text_hint_collection_size;
    }
}

File: app/src/main/java/com/example/task2/kotlinfolder/viewviewmodel/fragments/startingfragments/collectionsstarting/CollectionsStartingFragment.kt
package com.example.task2.kotlinfolder.viewviewmodel.fragments.startingfragments.collectionsstarting

import com.example.task2.R
import com.example.task2.kotlinfolder.viewviewmodel.fragments.BaseViewModel
import com.example.task2.kotlinfolder.viewviewmodel.fragments.startingfragments.ParentStartingFragment
import org.koin.androidx.viewmodel.ext.android.viewModel

class CollectionsStartingFragment : ParentStartingFragment() {

    override val viewModel: CollectionsStartingViewModel by viewModel()

    override val nextFragmentId: Int get() = R.id.action_collectionsStartingFragment_to_fragmentWithCollectionsBenchmarks

    override val stringIdForTextView: Int get() = R.string.enter_collection_size

    override val stringIdForEditTextHint: Int get() = R.string.edit_text_hint_collection_size
}